core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -1164,10 +1117,10 @@ public int length() {
         public boolean next() {
             while (true) {
                 edgeId++;
-                edgePointer = (long) edgeId * edgeAccess.getEntryBytes();
                 if (edgeId >= baseGraph.edgeCount)
                     return false;
 
+                edgePointer = edgeAccess.toPointer(edgeId);
```

Comment:
Should we not update also the edgePointer to the new value before returning?

Reply:
Hm, if `false` is returned we are not using the iterator anymore? But maybe you are right and its more consistent the way you proposed. Should not matter performance-wise, because usually it has to be updated anyway (unless we are at the end of the iteration)

Reply:
But then we need to update adjNode and baseNode etc. as well?

Reply:
Ah, indeed, you are right.